analysis/wwz/wwz4l.py

```diff
@@ -745,71 +784,82 @@ def process(self, events):
+                selections.add("sr_4l_bdt_of_3", (sr_4l_bdt_of_trn & bdt_of_3))
+                selections.add("sr_4l_bdt_of_4", (sr_4l_bdt_of_trn & bdt_of_4))
+                selections.add("sr_4l_bdt_of_5", (sr_4l_bdt_of_trn & bdt_of_5))
+                selections.add("sr_4l_bdt_of_6", (sr_4l_bdt_of_trn & bdt_of_6))
+                selections.add("sr_4l_bdt_of_7", (sr_4l_bdt_of_trn & bdt_of_7))
+                selections.add("sr_4l_bdt_of_8", (sr_4l_bdt_of_trn & bdt_of_8))
+
+                bdt_sr_names = [
+                    "sr_4l_bdt_sf_1",
+                    "sr_4l_bdt_sf_2",
+                    "sr_4l_bdt_sf_3",
+                    "sr_4l_bdt_sf_4",
+                    "sr_4l_bdt_sf_5",
+                    "sr_4l_bdt_sf_6",
+                    "sr_4l_bdt_sf_7",
+
+                    "sr_4l_bdt_of_1",
+                    "sr_4l_bdt_of_2",
+                    "sr_4l_bdt_of_3",
+                    "sr_4l_bdt_of_4",
+                    "sr_4l_bdt_of_5",
+                    "sr_4l_bdt_of_6",
+                    "sr_4l_bdt_of_7",
+                    "sr_4l_bdt_of_8",
+                ]
+                bdt_misc_names = [
+                    "sr_4l_bdt_sf_presel",
+                    "sr_4l_bdt_sf_trn",
+                    "sr_4l_bdt_of_trn",
+                    "sr_4l_bdt_of_presel",
+                ]
 
             cat_dict = {
                 "lep_chan_lst" : [
                     "sr_4l_sf_A","sr_4l_sf_B","sr_4l_sf_C","sr_4l_of_1","sr_4l_of_2","sr_4l_of_3","sr_4l_of_4",
                     "all_events","4l_presel", "sr_4l_sf_incl", "sr_4l_of_incl",
                     "cr_4l_btag_of", "cr_4l_btag_sf_offZ_met80", "cr_4l_sf",
-                    "sr_4l_bdt_sf_presel", "sr_4l_bdt_sf_trn", "sr_4l_bdt_of_trn", "sr_4l_bdt_of_presel",
-                ] + bdt_sr_names
+                ]
             }
 
+            if "2022" not in year:
+                cat_dict["lep_chan_lst"] = cat_dict["lep_chan_lst"] + bdt_sr_names + bdt_misc_names
+
             ######### Fill histos #########
 
             hout = {}
 
 
             # List the hists that are only defined for some categories
-            analysis_cats = ["sr_4l_sf_A","sr_4l_sf_B","sr_4l_sf_C","sr_4l_of_1","sr_4l_of_2","sr_4l_of_3","sr_4l_of_4"] + bdt_sr_names
+            analysis_cats = ["sr_4l_sf_A","sr_4l_sf_B","sr_4l_sf_C","sr_4l_of_1","sr_4l_of_2","sr_4l_of_3","sr_4l_of_4"]
+            if "2022" not in year:
+                analysis_cats = analysis_cats + bdt_sr_names
             exclude_var_dict = {
                 "mt2" : ["all_events"],
                 "ptl4" : ["all_events"],
-                "j0pt" : ["all_events", "4l_presel", "sr_4l_sf_incl", "sr_4l_of_incl", "sr_4l_bdt_sf_presel", "sr_4l_bdt_sf_trn", "sr_4l_bdt_of_trn", "sr_4l_bdt_of_presel", "cr_4l_sf"] + analysis_cats,
+                "j0pt" : ["all_events", "4l_presel", "sr_4l_sf_incl", "sr_4l_of_incl", "cr_4l_sf"] + analysis_cats,
```

Comment:
We still need to exclude the SR categories here, right?

Reply:
yes, but this is done on line 927 for when the year is not 2022

Reply:
Ok. I will try to think about how to make this part of the code less confusing (I don't like how I'm checking which histograms make sense for which categories). Anyway, this is fine for now I think.